### ISSUE TITLE
Error checking if CosmosDB Account 'XX' already exists

### ISSUE THREAD

**Issue by KenticoMartinS** [terraform-providers/terraform-provider-azurerm#3739]
### Terraform (and AzureRM Provider) Version
* `Terraform v0.11.14`
* `provider.azurerm: version = "~> 1.30.1"`

### Affected Resource(s)

* `azurerm_cosmosdb_account`

### Terraform Configuration Files

We are using standard `azurerm_cosmosdb_account` resource and everything was working till this morning.

### Expected Behavior

Cosmos DB account is created.

### Actual Behavior

Apply fails with an error:
```
 azurerm_cosmosdb_account.cosmosdb_account: 1 error occurred:
 azurerm_cosmosdb_account.cosmosdb_account: Error checking if CosmosDB Account "xxx-ab-745dr" already exists (Resource Group "RG123_745dr_AB"): documentdb.DatabaseAccountsClient#CheckNameExists: Failure responding to request: StatusCode=500 -- Original Error: autorest/azure: error response cannot be parsed: "" error: EOF
```

**Comment by MarkDordoy** [comment]
Im also seeing this issue. My configuration was working fine yesterday and today im also getting the error. As im still testing terraform i have deleted my CosmosDB, deleted my state file and deleted .terraform directory, re-ran `terroform init`  then on `terraform apply` it errors. Details below

**Terraform (and AzureRM Provider) Version**
* `Terraform v0.12.3`
* `provider.azurerm: version = "= 1.30.1"`

Error:
```
Error: Error checking if CosmosDB Account "mycosomsdbaccountname" already exists (Resource Group "myresourcegroupname"): documentdb.DatabaseAccountsClient#CheckNameExists: Failure responding to request: StatusCode=500 -- Original Error: autorest/azure: error response cannot be parsed: "" error: EOF

  on cosmosdb.tf line 2, in resource "azurerm_cosmosdb_account" "trade_cosmosdb_account_single_region":
   2: resource "azurerm_cosmosdb_account" "trade_cosmosdb_account_single_region" {
```

**Comment by avnes** [comment]
I get the same error too. It was working as late as yesterday.

```
Error: Error checking if CosmosDB Account "<account_name>" already exists (Resource Group "<group_name>"): documentdb.DatabaseAccountsClient#CheckNameExists: Failure responding to request: StatusCode=500 -- Original Error: autorest/azure: error response cannot be parsed: "" error: EOF

  on ../modules/cosmosdb_account/main.tf line 1, in resource "azurerm_cosmosdb_account" "db":
   1: resource "azurerm_cosmosdb_account" "db" {
```

```
terraform version
Terraform v0.12.3
+ provider.azurerm v1.29.0
```

I have also seen the same issue today with provider.azurerm v1.30.1

I was upgrading to Terraform 0.12.3 from 0.12.2 either today or yesterday, so I will attemt an downgrade to Terraform 0.12.2 to see if that helps. Will keep you posted.

**Comment by dan-v-elevate** [comment]
I'm seeing same issue on terraform v0.11.x as of yesterday as well.

**Comment by mheggeseth** [comment]
FYI, no details yet, but Microsoft Support has acknowledged this problem and the the "Cosmos DB Engineering Team" is working on resolving it.

**Comment by RomanKreisel** [comment]
Is there any URL from microsoft known, where this issue can be tracked?

**Comment by tombuildsstuff** [comment]
👋 

As mentioned above this is a bug in the API which is sounds like the CosmosDB team are looking into - however in the interim [we've raised a bug about it here](https://github.com/Azure/azure-sdk-for-go/issues/5157). 

Until this is fixed since this functionality only serves to raise a more helpful error - we've opened #3747 which will temporarily ignore a 500 being returned from this API - so that the CosmosDB resources can still be provisioned (but may return a less helpful error if they conflict with an existing resource) - which will ship in the 1.31 release later today.

Thanks!
